src/OrchardCore.Modules/OrchardCore.OpenId/Recipes/OpenIdScopeStep.cs

```diff
@@ -27,20 +28,29 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
                 return;
             }
 
-            var model = context.Step.ToObject<OpenIdScopeStepViewModel>();
+            var model = context.Step.ToObject<OpenIdScopeStepModel>();
+
             var descriptor = new OpenIdScopeDescriptor
             {
                 Description = model.Description,
                 Name = model.ScopeName,
                 DisplayName = model.DisplayName
             };
 
-            if (!string.IsNullOrEmpty(model.Resources))
+            if (model.Resources.Any())
             {
-                descriptor.Resources.UnionWith(model.Resources.Split(new[] { " " }, StringSplitOptions.RemoveEmptyEntries));
+                descriptor.Resources.UnionWith(model.Resources);
             }
 
-            await _scopeManager.CreateAsync(descriptor);
+            var scope = await _scopeManager.FindByNameAsync(model.ScopeName);
+            if (scope != null)
+            {
+                await _scopeManager.UpdateAsync(scope, descriptor);
```

Comment:
Same remark, the current code is not suitable for UPSERT scenarios.

Reply:
I get you're point have been looking at the code. Did test this easier solution and it seems to work.
At least, I expect that executing a recipe does a reset of the scope. This code resets the scope to the value's of the recipe.
My solution starts with an empty descriptor and then adds the values of the model. The controller starts with a descriptor based on the model, resets it and then fills it with the values of the model.

Reply:
Do I need to add explicit the current tenant to the resources, as is done in the controller? Was not sure about that.

Reply:
The problem with starting with an empty descriptor is that every custom permission/property won't be preserved as the recipe won't re-add it. It would be better to populate the descriptor and amend it as in the app/scopes controller.

Reply:
Ok, will do that, what about the current tenant, which is added in the scope controller

Reply:
That's pretty much the opposite: the scope controller filters out the current tenant as it's always included later by the authorization controller: https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/ScopeController.cs#L145

That said, I think it would be nice to also have a `Tenants` property in the recipe and use the same logic.

Reply:
Added a TentantNames property to the recipe